### PR TITLE
[Quant] [PT2] Add EmbeddingBag  into X86InductorQuantizer Annotation

### PR DESCRIPTION
**Pull request by zhuhaozhe**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122488

### CODE REVIEW

File: test/quantization/pt2e/test_x86inductor_quantizer.py [hunk 1]
```diff
@@ -304,6 +304,20 @@ def forward(self, x):
             weighted = torch.bmm(attention, v)
             return weighted
 
+    class EmbeddingBagList(torch.nn.Module):
+        def __init__(self, ntables=3, embedding_dim=128, num_embedings=1000) -> None:
```
Comment (jgong5) [inline test/quantization/pt2e/test_x86inductor_quantizer.py:308]: nit:
```suggestion
        def __init__(self, ntables=3, embedding_dim=128, num_embeddings=1000) -> None:
```

**Review by jgong5 — CHANGES_REQUESTED** [review]
tests are failing.

File: test/quantization/pt2e/test_x86inductor_quantizer.py (hunk 1 above)
Comment (zhuhaozhe) [reply test/quantization/pt2e/test_x86inductor_quantizer.py]: Thanks for reminder

**Comment by zhuhaozhe** [comment]
> tests are failing.

I should use `elif` but used `if', fixed.

File: torch/ao/quantization/quantizer/x86_inductor_quantizer.py [hunk 2]
```diff
@@ -878,6 +898,8 @@ def is_all_inputs_connected_to_quantized_op(input_nodes):
                 if not is_all_inputs_connected_to_quantized_op(input_nodes_to_check):
                     return
                 self._annotate_cat(node, quantization_config)
+            elif node.target is torch.ops.aten.embedding_bag.padding_idx:
```
Comment (leslie-fang-intel) [inline torch/ao/quantization/quantizer/x86_inductor_quantizer.py:901]: `_annotation_propagation_quantizable_pattern` is used to annotate operators like `maxpool2d`, `cat` which will depend on previous nodes to decide whether we should insert observer for it. But here, the recipe for `embedding_bag` is insert observer at weight no matter of previous node.

- Can we double check in IPEX whether it's a desired recipe?
- If so, I think the implementation of recipe for `embedding_bag` shouldn't in `_annotation_propagation_quantizable_pattern` but somewhere like the conv/linear.

**Review by leslie-fang-intel — CHANGES_REQUESTED** [review]

File: torch/ao/quantization/quantizer/x86_inductor_quantizer.py [hunk 3]
```diff
@@ -68,10 +68,14 @@ class _X86InductorQuantizationAnnotation(QuantizationAnnotation):
...
 
-
+embedding_ops: Set = {
+    torch.ops.aten.embedding_bag.padding_idx,
+}
 # Operations support the int8 data type and exclude operations such as conv and linear.
 # A superset of int8_in_int8_out_ops_pt2e incorporating additional operators.
-quantizable_ops_pt2e = copy.deepcopy(int8_in_int8_out_ops_pt2e)
+quantizable_ops_pt2e = copy.deepcopy(int8_in_int8_out_ops_pt2e).union(
+    copy.deepcopy(embedding_ops)
```
Comment (leslie-fang-intel) [inline torch/ao/quantization/quantizer/x86_inductor_quantizer.py:77]: If we always annotate weight of `embedding_ops`. We don't need to add it into `quantizable_ops_pt2e ` as the comment above.

**Pull request closed without merging**